packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/EnumerableSet.sol";
+
+import "./UsingRegistry.sol";
+import "../common/Freezable.sol";
+import "../common/FixidityLib.sol";
+import "../common/Initializable.sol";
+
+import "../common/interfaces/IFeeHandler.sol";
+import "../common/interfaces/IFeeHandlerSeller.sol";
+
+// TODO move to IStableToken when it adds method getExchangeRegistryId
+import "../stability/StableToken.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+import "../common/interfaces/ICeloToken.sol";
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+
+// Using the minimal required signatures in the interfaces so more contracts could be compatible
+import "../uniswap/interfaces/IUniswapV2RouterMin.sol";
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+contract FeeHandler is
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ICeloVersionedContract,
+  Freezable,
+  IFeeHandler,
+  ReentrancyGuard
+{
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+  uint256 public constant FIXED1_UINT = 1000000000000000000000000;
```

Comment:
Do we need this constant? If so we could consider extending FixidityLib to expose it

Reply:
mmm, updating FixidityLib will trigger redeploys for everything